### PR TITLE
Supporting minimal terraform verbosity for running with LLM pipeline.

### PR DESCRIPTION
**Pull request by soumyapani**

### CODE REVIEW

File: scripts/_terraform_util.sh [hunk 1]
```diff
@@ -19,7 +19,15 @@
 #
 _terraform_setup() {
     apply_ret=0
-    terraform -chdir=/usr/primary apply -input=false -auto-approve || apply_ret=$?
+
+    # change terraform verbosity based on SHOW_PROXY_URL environment variable.
+    if [[ ! -z "$SHOW_PROXY_URL" && "${SHOW_PROXY_URL,,}" == "no" ]]; then
```
Comment (abdallag) [inline scripts/_terraform_util.sh:24]: You can use -n instead of '! -z'

File: scripts/_terraform_util.sh [hunk 2]
```diff
@@ -78,7 +86,15 @@ _terraform_cleanup() {
             export IS_CLEANUP_NEEDED="no"
             echo "Calling terraform destroy..."
             destroy_ret=0
-            terraform -chdir=/usr/primary destroy -input=false -auto-approve || destroy_ret=$?
+
+            # change terraform verbosity based on SHOW_PROXY_URL environment variable.
```
Comment (abdallag) [inline scripts/_terraform_util.sh:90]: It would be better to have a separate environment variable to control logging verbosity

File: README.md [hunk 3]
```diff
@@ -34,7 +34,8 @@ The optional parameters are:
     - The `<NFS fileshare type>` cannot be empty. The supported values are `BASIC_HDD`,`BASIC_SSD`,`HIGH_SCALE_SSD` and `ENTERPRISE`.
     - The `<NFS fileshare size in GB>` can be empty and the default value is 2560 GB (2.5 TB).
 18. ***SHOW_PROXY_URL***. This controls if the Jupyter notebook proxy url is retrieved for the cluster or not. The default value is yes. If this is present and set to no, then connection information is not collected. The supported values are: yes, no.
-19. ***NETWORK_CONFIG***. This controls the VPC type to be used for the MIG. The supported values are default_network, new_network and multi_nic_network. The dault value is default_network. The behaviour is 
+19. ***MINIMIZE_TERRAFORM_LOGGING***. This controls the verbosity of terraform logs. When any value is set for this parameter, the terraform output is reditrected to a local file and not printed on syserr. The log file is then uploaded to storage account. Any value can be set for this parameter, e.g.: yes, true.
```
Comment (DmitryKakurin) [inline README.md:37]:
```suggestion
19. ***MINIMIZE_TERRAFORM_LOGGING***. This controls the verbosity of terraform logs. When any value is set for this parameter, the terraform output is redirected to a local file and not printed on syserr. The log file is then uploaded to storage account. Any value can be set for this parameter, e.g.: yes, true.
```

File: scripts/_terraform_util.sh [hunk 4]
```diff
@@ -78,7 +86,15 @@ _terraform_cleanup() {
             export IS_CLEANUP_NEEDED="no"
             echo "Calling terraform destroy..."
             destroy_ret=0
-            terraform -chdir=/usr/primary destroy -input=false -auto-approve || destroy_ret=$?
+
+            # change terraform verbosity based on MINIMIZE_TERRAFORM_LOGGING environment variable.
+            if [[ -n "$MINIMIZE_TERRAFORM_LOGGING" ]]; then
+                echo "Redirecting terraform destroy output to $TERRAFORM_LOG_PATH."
```
Comment (DmitryKakurin) [inline scripts/_terraform_util.sh:92]:
```suggestion
                echo "Redirecting 'terraform destroy' output to $TERRAFORM_LOG_PATH."
```

File: scripts/_terraform_util.sh [hunk 5]
```diff
@@ -134,6 +151,11 @@ _perform_terraform_action() {
     else
         echo "Action $ACTION is not supported..."
     fi
+
+    if [ -f "$TERRAFORM_LOG_PATH" ]; then
+        echo -e "${GREEN}Copying terraform output file from $TERRAFORM_LOG_PATH ${NOC}"
+        gsutil -m cp $TERRAFORM_LOG_PATH gs://$TF_BUCKET_NAME/$TF_DEPLOYMENT_PATH/$NAME_PREFIX-terraformlog.txt
```
Comment (DmitryKakurin) [inline scripts/_terraform_util.sh:157]: Perhaps `terraform.log` for a file name? log.txt seems redundant.

**Review by DmitryKakurin — CHANGES_REQUESTED** [review]

**Review by abdallag — APPROVED** [review]

**Review by DmitryKakurin — APPROVED** [review]

**Pull request merged**